### PR TITLE
Fix scanner object leaks close them appropriately.

### PR DESCRIPTION
**Pull request by harshavardhana**
Currently scanner was never closed which would leak fds.

### CODE REVIEW

File: api/src/main/java/io/minio/MinioClient.java [hunk 1]
```diff
@@ -967,6 +965,7 @@ private HttpResponse execute(Method method, String region, String bucketName, St
       } finally {
         response.body().close();
       }
+      scanner.close();
```
Comment (balamurugana) [inline api/src/main/java/io/minio/MinioClient.java]: Adding `scanner.close();` in `finally` is good enough than moving scanner out of `try` block.

File: api/src/main/java/io/minio/MinioClient.java [hunk 2]
```diff
@@ -2938,20 +2937,18 @@ private void completeMultipart(String bucketName, String objectName, String uplo
...
-    } catch (EOFException e) {
-      // Getting EOF exception is not an error.
-      // Just log it.
-      LOGGER.log(Level.WARNING, "EOF exception occured: " + e);
-    }  finally {
+    } finally {
       response.body().close();
     }
-
+    scanner.close();
```
Comment (balamurugana) [inline api/src/main/java/io/minio/MinioClient.java]: same as above

File: api/src/main/java/io/minio/MinioClient.java (hunk 1 above)
Comment (harshavardhana) [inline api/src/main/java/io/minio/MinioClient.java]: Done.

**Comment by harshavardhana** [comment]
ping @balamurugana

File: api/src/main/java/io/minio/MinioClient.java [hunk 3]
```diff
@@ -950,11 +947,12 @@ private HttpResponse execute(Method method, String region, String bucketName, St
 
     // HEAD returns no body, and fails on parseXml
     if (!method.equals(Method.HEAD)) {
+      Scanner scanner = new Scanner(response.body().charStream());
+      scanner.useDelimiter("\\A");
```
Comment (balamurugana) [inline api/src/main/java/io/minio/MinioClient.java]: You missed scanner to be moved inside `try` block.

File: api/src/main/java/io/minio/MinioClient.java [hunk 4]
```diff
@@ -2938,18 +2937,16 @@ private void completeMultipart(String bucketName, String objectName, String uplo
 
     // Fixing issue https://github.com/minio/minio-java/issues/391
     String bodyContent = "";
+    Scanner scanner = new Scanner(response.body().charStream());
```
Comment (balamurugana) [inline api/src/main/java/io/minio/MinioClient.java]: same as above

File: api/src/main/java/io/minio/MinioClient.java (hunk 3 above)
Comment (harshavardhana) [inline api/src/main/java/io/minio/MinioClient.java]: It won't work @balamurugana there is a compilation error..

Comment (harshavardhana) [inline api/src/main/java/io/minio/MinioClient.java]: so what is done right now is the correct style..

File: api/src/main/java/io/minio/MinioClient.java (hunk 4 above)
Comment (harshavardhana) [inline api/src/main/java/io/minio/MinioClient.java]: Read the above comment.. try doesn't work.

**Comment by harshavardhana** [comment]
ping @balamurugana

**Pull request merged**